### ISSUE TITLE
Aggregator tracker

### ISSUE THREAD

**Issue by ValerianRey** [TorchJD/torchjd#665]
This issue tracks some candidate aggregators that could maybe be added to TorchJD. Feel free to suggest other methods, or to add more information about the methods in the table. If you're interested in implementing one, please tell it so that we can investigate the method in a bit more details (in particular, we need to verify that it can be implemented as an aggregator), and so that we don't work concurrently on the same thing.

Status | Name | Ref | Stateful | Existing implementations | Special Remarks |
|-|-|-|-|-|-|
🟡 In progress (#676) | MoCo (Multi-Objective Gradient Correction) | [Mitigating gradient bias in multi-objective learning](https://openreview.net/pdf?id=dLAYGdKTi2) (ICLR 2023 oral, 95 citations) | Yes (Jacobian EMA) | [LibMTL](https://github.com/median-research-group/LibMTL/blob/main/LibMTL/weighting/MoCo.py) | |
🟣 To Do | GradNorm | [GradNorm: Gradient Normalization for Adaptive Loss Balancing in Deep Multitask Networks](https://proceedings.mlr.press/v80/chen18a/chen18a.pdf) (ICML 2018, 2334 citations) | Yes, trainable state | [LucasBoTang (unofficial)](https://github.com/LucasBoTang/GradNorm/blob/main/gradnorm.py), [lucidrains (unofficial)](https://github.com/lucidrains/gradnorm-pytorch/blob/main/gradnorm_pytorch/gradnorm_pytorch.py), [LibMTL](https://github.com/median-research-group/LibMTL/blob/main/LibMTL/weighting/GradNorm.py)  | See discussion in scalarization channel on discord.
🟠 Too Early | DB-MTL (Dual Balancing Multi-Task Learning) | [Dual-balancing for multi-task learning](https://arxiv.org/pdf/2308.12029) (Neural Networks n°195, 9 citations) | Yes (Jacobian EMA) | [LibMTL (official)](https://github.com/median-research-group/LibMTL/blob/main/LibMTL/weighting/DB_MTL.py) | The paper combines 3 ideas: applying a log to the losses (not something we need TorchJD for), normalizing the gradients (this should be a Normalizer rather than a part of the aggregator) and using the EMA of the gradients. This last idea is also common with some other methods, so maybe we could implement it as a wrapper or something. So in the end I don't think there is anything to implement specifically for DB-MTL. We could make a class combining those ideas, but only after the normalization package is ready. |

**Comment by giovannicozzolongo** [comment]
I'd like to work on this. Could you assign it to me?

**Comment by ValerianRey** [comment]
> I'd like to work on this. Could you assign it to me?

Sure! Which aggregator would you like to work on? Also, please join our discord server to discuss implementation details (we've already talked quite a bit about those in the table)

https://discord.gg/76KkRnb3nk

**Comment by giovannicozzolongo** [comment]
Thanks! I've joined and I'd like to work on GradNorm
